### DISCUSSION TITLE
geopandas.GeoDataFrame.clip returns empty clippings

### DISCUSSION THREAD

**Discussion by TarunSrinivas23** [geopandas/geopandas#2946] [Q&A] [answered]
Hey, I am trying to use GeoPandas to clip a few LiDAR pointclouds (.LAS files) and their respective polygons (.SHP files). My approach is I read them using laspy and save them as a geopandas.GeoDataFrame, after which I use geopandas to read the polygon, set CRS and then use clip. I don't know what is going wrong, I even set the offsets and scales, it still returns an empty result.
 
```
import laspy
import numpy as np
import geopandas as gpd
las_file = laspy.read('cropped_las.las')

points = gpd.GeoDataFrame(geometry=gpd.points_from_xy(las_file.x, las_file.y))
points.crs = "EPSG:26986"
# Read the shapefile
polygons = gpd.read_file("structures_poly_9.shp")
print(type(polygons))
polygons.crs = "EPSG:26986"
clipped_points = points.clip(polygons)

print(clipped_points.shape)
```

Any help regarding this issue? I am not sure if it is a bug or a problem with how I am approaching this solution. Thank you in advance!!

**Comment by martinfleis** [comment, marked as answer]
Do not assign directly to the `.crs` attribute. Use `set_crs` or `to_crs` instead. You want to reproject `polygons` using `to_crs` here to match the one of points. Not just override the information about the CRS with no change to actual geometries.

**Reply by TarunSrinivas23** [reply]
Thanks for the quick reply ! So, the .las file did not have any CRS initially (that might be something to do from reading with laspy). But the shapefile containing polygons did. I tried setting the CRS from the polygons to the points, like below. But I still got an empty list as output.. Any idea on why this could be happening or debug strategies? Thanks again for the support.
```python
las_file = laspy.read('cropped_las.las')
# print(las_file.header.offsets)
polygons = gpd.read_file("structures_poly_9.shp")
points = gpd.GeoDataFrame(geometry=gpd.points_from_xy(las_file.x, las_file.y), crs=polygons.crs)
points.to_crs(polygons.crs, inplace=True)
clipped_points = points.clip(polygons)

print(clipped_points.shape)
```

**Reply by martinfleis** [reply]
As I said, you need to use to_crs on polygons. Not points.

**Reply by martinfleis** [reply]
You also need to ensure that you are assigning a correct CRS to points created from your las file.

**Reply by TarunSrinivas23** [reply]
Yeah that makes sense. I think the issue lies in the CRS for the points from the las file. Do you have any suggestions on how to read a LAZ file from USGS the right way for use for clipping? I am new to this, so I used laspy. Any other way I can do this? I am using Ubuntu 22.04 and Python 3.10

**Reply by martinfleis** [reply]
I'm not familiar with this type of data, sorry.
